src/puppetlabs/puppetdb/http/fact_names.clj

```diff
@@ -1,27 +1,40 @@
 (ns puppetlabs.puppetdb.http.fact-names
   (:require [puppetlabs.puppetdb.query.facts :as f]
+            [puppetlabs.puppetdb.query-eng :refer [produce-streaming-body]]
             [puppetlabs.puppetdb.query.paging :as paging]
             [puppetlabs.puppetdb.jdbc :refer [with-transacted-connection]]
+            [puppetlabs.puppetdb.utils :refer [assoc-when]]
             [net.cgrand.moustache :refer [app]]
-            [puppetlabs.puppetdb.middleware :refer [verify-accepts-json validate-query-params wrap-with-paging-options validate-no-query-params]]
+            [puppetlabs.puppetdb.middleware :refer [verify-accepts-json
+                                                    validate-query-params
+                                                    wrap-with-paging-options]]
             [puppetlabs.puppetdb.http :refer [query-result-response]]))
 
-(defn get-fact-names
-  "Produces a response body containing the list of known facts."
-  [{:keys [globals paging-options]}]
-  (let [db (:scf-read-db globals)
-        facts (with-transacted-connection db
-                (f/fact-names paging-options))]
-    (query-result-response facts)))
+(defn query-app
+  ([version]
+   (app
+     [&]
+     {:get (fn [{:keys [params globals paging-options]}]
+             (let [paging-options (assoc-when paging-options
+                                              :order_by [[:name :ascending]])]
+               (produce-streaming-body
+                 :fact-names
+                 version
+                 (params "query")
+                 paging-options
```

Comment:
We're calling this `query-options` elsewhere, should we consolidate on one of the two?